### PR TITLE
Cupertino demo improvements

### PR DESCRIPTION
**Pull request by guidezpl**
Easy fixes to cupertino demos:
- adding disabled examples to buttons and switches
- removing the pull to refresh demo. It's not working on web (intentionally), and is an outlier as it isn't a component. Since the sample code for [`CupertinoSliverRefreshControl`](https://api.flutter.dev/flutter/cupertino/CupertinoSliverRefreshControl-class.html) is more complete, remove this demo

Remaining issues are in the framework (e.g. flutter/flutter#54241)

Additonal change: I made the test which verifies `code_segments.dart` is up to date less verbose in [CI](https://github.com/flutter/gallery/runs/6223366321?check_suite_focus=true), and make it just show the error instead. The full diff took forever to render and hid the actual error.

### CODE REVIEW

**Comment by pennzht** [comment]
Thanks for the PR!
It looks like the goldens need replacing.

**Comment by guidezpl** [comment]
Yeah the goldens have been failing for a while. I was waiting for things to settle upstream (low-level rendering changes). I'll get new goldens in.

File: lib/demos/cupertino/cupertino_picker_demo.dart [hunk 1]
```diff
@@ -63,13 +63,15 @@ class _CupertinoPickerDemoState extends State<CupertinoPickerDemo> {
...
-        ]),
+        child: _Menu(
+          children: [
+            Text(GalleryLocalizations.of(context)!.demoCupertinoPickerDate),
+            Text(
+              DateFormat.yMMMMd().format(date),
+              style: const TextStyle(color: CupertinoColors.inactiveGray),
+            ),
+          ],
+        ),
```
Comment (pennzht) [inline lib/demos/cupertino/cupertino_picker_demo.dart:74]: Is this just a change in formatting (indentation)?

File: test/code_segments_test.dart [hunk 2]
```diff
@@ -8,12 +8,15 @@ import '../tool/codeviewer_cli/segment_generator.dart';
...
+    return standardizeLineEndings(a) == standardizeLineEndings(b);
+  }
+
   test('verify code segments are up to date', () async {
     final currentCodeSegments = readCodeSegments();
     var newCodeSegments = await getCodeSegments();
 
-    expect(standardizeLineEndings(newCodeSegments),
-        standardizeLineEndings(currentCodeSegments),
+    expect(_compareCodeSegments(currentCodeSegments, newCodeSegments), true,
```
Comment (pennzht) [inline test/code_segments_test.dart:19]: I like how this change simplifies the log!

**Review by pennzht — APPROVED** [review]
LGTM. Thank you!

File: lib/demos/cupertino/cupertino_picker_demo.dart (hunk 1 above)
Comment (guidezpl) [reply lib/demos/cupertino/cupertino_picker_demo.dart:74]: exactly

**Pull request merged**